ts/src/pro/binance.ts

```diff
@@ -830,15 +839,10 @@ export default class binance extends binanceRest {
         const symbol = market['symbol'];
         const messageHash = 'orderbook::' + symbol;
         if (!(symbol in this.orderbooks)) {
-            //
+            // Sometimes Binance sends the first delta before the subscription confirmation arrives
             // https://github.com/ccxt/ccxt/issues/6672
-            //
-            // Sometimes Binance sends the first delta before the subscription
-            // confirmation arrives. At that point the orderbook is not
-            // initialized yet and the snapshot has not been requested yet
-            // therefore it is safe to drop these premature messages.
-            //
-            return;
+            const subscription = this.safeValue (client.subscriptions, messageHash, {});
+            this.tryInitializeOrderBook (symbol, subscription);
```

Comment:
It does not hurt to cache this messages, but the issue is I believe we should reset the `this.orderbook` whenever a subscriptions is confirmed or created